### ISSUE TITLE
[Docs] Make it clear what dependencies should be imported when using @babel/preset-env together with useBuiltIns options

### ISSUE THREAD

**Issue by wujekbogdan** [babel/website#2163]
Hi,

I've read the [`@babel/preset-env` documentation](https://babeljs.io/docs/en/babel-preset-env) several times, but I'm still not 100% sure If I get it all correctly.

To me, it seems that:

- when `@babel/preset-env` is used together with `useBuiltIns: 'usage'` then no additional dependencies need to be imported in our entry file. Babel will pick and import all the polyfills for us based on features used in the transpiled code (and based on an additional configuration like browserslist config or `target` config option).
- when we use `useBuiltIns: 'entry'` then we need to add the following lines to our entry file:
    ```
    import 'core-js/stable';
    import 'regenerator-runtime/runtime';
    ```
- when we use `useBuiltIns: false` then we have to import the entire `@babel/polyfill` or add import individual polyfills.

Both: `usage` and `entry` options require the `core-js` lib to be installed.

---
Could you improve the docs and make it really clear? Some examples would be very helpful.

**Comment by danBamikiya** [comment]
`[@babel/polyfill](https://babeljs.io/docs/en/babel-polyfill)` has been deprecated.

**Comment by nicolo-ribaudo** [comment]
Yeah, because it was just a wrapper around `core-js@2` and `regenerator-runtime`.
We decided to let users include these two dependencies themselves so that they can control the version (and we don't take the merits of `core-js`).
